am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/defaultrolesetup/DefaultPermissionIntegrationTest.java

```diff
@@ -0,0 +1,107 @@
+package integration.uk.gov.hmcts.reform.amlib.defaultrolesetup;
+
+import integration.uk.gov.hmcts.reform.amlib.base.IntegrationBaseTest;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.amlib.enums.AccessType;
+import uk.gov.hmcts.reform.amlib.enums.RoleType;
+import uk.gov.hmcts.reform.amlib.enums.SecurityClassification;
+import uk.gov.hmcts.reform.amlib.exceptions.PersistenceException;
+import uk.gov.hmcts.reform.amlib.models.DefaultPermissionGrant;
+import uk.gov.hmcts.reform.amlib.utils.Permissions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createDefaultPermissionGrant;
+import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createReadPermissionsForAttribute;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ATTRIBUTE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.CREATE_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.READ_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class DefaultPermissionIntegrationTest extends IntegrationBaseTest {
+
+    @Test
+    void shouldNotBeAbleToCreateDefaultPermissionWhenRoleDoesNotExist() {
+        defaultRoleService.deleteRole(ROLE_NAME);
+
+        assertThatExceptionOfType(PersistenceException.class)
+            .isThrownBy(() -> defaultRoleService.grantDefaultPermission(createDefaultPermissionGrant(READ_PERMISSION)))
+            .withMessageContaining("(role_name)=(Role Name) is not present in table \"roles\"");
+    }
+
+    @Test
+    void whenAddResourceDefinitionIsCalledAddNewEntryIntoDatabase() {
```

Comment:
Ideally we should try to use same format for test names. I like "shouldDoXWhenY" format for example.

I won't be pushing it this time thought ;)

Reply:
The name does not look right. Method tests `grantDefaultPermission` but it says `whenAddResourceDefinition`.